### DISCUSSION TITLE
set ssh config path

### DISCUSSION THREAD

**Discussion by aserowy** [wezterm/wezterm#1146] [Q&A]
Heho,

till wsl 2 does not support sockets to connect directly im using ssh to connect from my windows to the ssl instance (thus i get persistence even when i close my frontend). Normally my whole workflow with ssh lies in my linux. (windows just for productivity tools and .net fullframework projects. dont ask, its another story :).)

To be able to use ssh with keys i have to setup keys on my windows as well. (and it really bugs me). With an option in wezterm.lua to set the ssh directory for wezterm, i could reference into my wsl 2 directory and would only had one place to manage keys. What would be awesome!

Or am i missing some options, which would support the idea already?

Kind regards
Alexander

**Comment by wez** [comment]
Are you using `wezterm ssh` or using ssh multiplexing?

We don't currently have a way to specify an alternative `~/.ssh/config` path, but I'm not sure that that is exactly what you want, because a lot of things around ssh config parsing are resolved based on the home directory and username, and the home directly is definitely a different path on the host Windows system and the target WSL system.

It is possible to pass in ssh option overrides when running `wezterm ssh`; this is mentioned in https://wezfurlong.org/wezterm/ssh.html#cli-overrides which shows how to use a specific key file, which in your situation could reference the key path inside your wsl 2 directory.

We could potentially make it a bit more convenient to specify similar overrides in the wezterm config, but depending on whether you are using ad-hoc ssh or multiplexing, the implementation is a little different.

**Reply by aserowy** [reply]
Running ssh for multi plexing. Config in wezterm.lua would be golden!

Kind regards

PS. Instead of ad hoc in my work flow I would use wsl.exe instead

**Reply by wez** [reply]
I've filed https://github.com/wez/wezterm/issues/1149 to track this. I'm not sure when I'll get to it: I've got a big move coming up over the next month, but at least it won't be forgotten!

**Reply by aserowy** [reply]
> I've filed https://github.com/wez/wezterm/issues/1149 to track this. I'm not sure when I'll get to it: I've got a big move coming up over the next month, but at least it won't be forgotten!

Best wishes to your move and ty for the heads up!
